src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -389,9 +389,60 @@ public void Initialize()
                 }
             }
 
+            using (ContextPool.AllocateOperationContext(out TransactionOperationContext ctx))
+            {
+                if (TryLoadAuthenticationKeyPairs(ctx) == false)
+                    GenerateAuthenticationKeyPairs(ctx);
+            }
+
             Task.Run(ClusterMaintenanceSetupTask, ServerShutdown);
         }
 
+        public byte[] BoxPublicKey, BoxSecretKey, AuthPublicKey, SignSecretKey;
+
+        private void GenerateAuthenticationKeyPairs(TransactionOperationContext ctx)
+        {
+            BoxPublicKey = new byte[Sodium.crypto_box_publickeybytes()];
+            BoxSecretKey = new byte[Sodium.crypto_box_secretkeybytes()];
+            AuthPublicKey = new byte[Sodium.crypto_sign_publickeybytes()];
```

Comment:
This should be called `SignPublicKey`, since it needs to match the reset of the params